xmodem/__init__.py

```diff
@@ -509,27 +531,12 @@ def recv(self, stream, crc_mode=1, retry=16, timeout=60, delay=1, quiet=0):
                     income_size += len(data)
                     stream.write(data)
                     self.putc(ACK)
+                    last_sequence = sequence
                     sequence = (sequence + 1) % 0x100
                     # get next start-of-header byte
                     char = self.getc(1, timeout)
                     continue
 
-            # something went wrong, request retransmission
-            self.log.warn('recv error: purge, requesting retransmission (NAK)')
-            while True:
-                # When the receiver wishes to <nak>, it should call a "PURGE"
-                # subroutine, to wait for the line to clear. Recall the sender
-                # tosses any characters in its UART buffer immediately upon
-                # completing sending a block, to ensure no glitches were mis-
-                # interpreted.  The most common technique is for "PURGE" to
-                # call the character receive subroutine, specifying a 1-second
-                # timeout, and looping back to PURGE until a timeout occurs.
-                # The <nak> is then sent, ensuring the other end will see it.
-                data = self.getc(1, timeout=1)
-                if data is None:
-                    break
-                assert False, data
```

Comment:
Interesting... this "assert False, data" looks like a pattern I use during development. One of us must have written this, hoping to express it by test case, and forgot to write whatever python code we thought was missing here :)